js/visualization_entity_tables_edit.js

```diff
@@ -1,13 +1,25 @@
 (function($) {
+
   $(document).on('ready', function(){
+
+    var settings = JSON.parse($('#edit-field-ve-settings-und-0-value').val());
```

Comment:
Here you are assuming that $('#edit-field-ve-settings-und-0-value').val() is an object, that was causing the end of input error. You need to check that before the parse.